### ISSUE TITLE
vertexColors are ignored in drawBuffersScaled

### ISSUE THREAD

**Issue by Vamoss** [processing/p5.js#3857]
#### Nature of issue?

- [ ] Found a bug
- [X] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [X] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [X] Mobile/Tablet (touch devices)
- [X] Desktop/Laptop
- [ ] Others (specify if possible)

#### Feature enhancement details:

![image](https://user-images.githubusercontent.com/245841/60617944-34a97500-9dab-11e9-8e3c-286ab71c3c5c.png)
When calling drawBuffersScaled, vertexColors are not being drew.

drawBuffersScaled calls drawBuffers(gId), witch not uses vertexColors, only the last fill color.
https://github.com/processing/p5.js/blob/3ec269b89dab5892104b2c0acadb0e82f988b2c3/src/webgl/p5.RendererGL.Retained.js#L209

I've taken a look at the beginShape() and endShape() functions, witch works with vertexColors, and this function calls this method:
https://github.com/processing/p5.js/blob/a8e19a30af0e5e53dd1570ac53a37446a80ee0ef/src/webgl/p5.RendererGL.Immediate.js#L280

Is it possible to make a standard way of drawing the geometry the same way of beginShape()?

**Comment by stalgiag** [comment]
Yes vertex colors should be relatively easy to add. As you have noticed a lot of the groundwork is already laid for them. That said, there are structural changes currently planned for immediate mode. Vertex color functionality should be simple to add but should wait until those structural changes are made. 

I am just going to tag this issue in that one to track this. 
related to #3620

**Comment by stalgiag** [comment]
@aferris - I can work on this eventually but it got put on the backburner since the person working on it left the repo. I think vertex colors would be nice though. We could enable vertex colors without the unification if you would like.

I can work on the unification sometime before the 1.0 release.

**Comment by stalgiag** [comment]
@Vamoss Hi I am just coming back around to this and reading your issue a bit more carefully and I have a bit of a misunderstanding about what is desired here. Can you show me a code example that you hope will work?

I am a bit confused because I don't know how per-vertex colors could possibly work with `drawBuffersScaled()` which is an optimization for retained geometry so that the hash can reuse similar geometry in different sizes. I am mostly confused because I don't know how you could specify per-vertex fills on the geometry primitives. This works currently for example:
```
  beginShape(TRIANGLE_FAN);
  fill(200, 100, 100);
  vertex(-200, 200, 0);
  fill(100, 200, 100);
  vertex(-200, -200, 0);
  fill(200, 100, 200);
  vertex(200, -200, 0);
  fill(100, 100, 200);
  vertex(200, 200, 0);
  endShape();
```
Assuming I understand correctly, the above translated into retained (which uses `drawBuffersScaled()` would be something like:
```
vertexColors.push(color(200, 100, 100));
vertexColors.push(color(100, 200, 100));
vertexColors.push(color(200, 100, 200));
vertexColors.push(color(100, 100, 200));
translate(-200, -200, 0);
plane(400, 400, 1, 1);
```
Am I correct in understanding that this is what you are looking for?

**Comment by Vamoss** [comment]
Hi @stalgiag, thank you for looking into this.

The code:
```js
function setup() {
	createCanvas(windowWidth, windowHeight, WEBGL);
}

function draw() {
	background(0);
	beginShape();
	fill(255, 0, 0);
	vertex(-100, -100);
	fill(255, 255, 0);
	vertex(100, -100);
	fill(255, 0, 255);
	vertex(100, 100);
	fill(0, 255, 255);
	vertex(-100, 100);
	endShape();
}
```

The result:
<img width="280" alt="Captura de Tela 2019-11-19 às 14 37 15" src="https://user-images.githubusercontent.com/245841/69166295-c2270080-0ad1-11ea-8637-7eff29fcac86.png">

Something changed since my initial text. I now see some vertex have colors, others not.

Cheers,
Carlos

**Comment by stalgiag** [comment]
Ah I see! The problem seems to be the same as #4141 which will be fixed by #3620 . I have a work-in-progress branch for this so expect a fix to appear soon.

**Comment by stalgiag** [comment]
Closing this as #3620 appears to have fixed the example given. Thanks!
